app/javascript/src/Plans/components/DefaultPlanSelectCard.jsx

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React, { useState } from 'react'
+import { post } from 'utilities/ajax'
+
+import {
+  Form,
+  FormGroup,
+  Card,
+  CardBody
+} from '@patternfly/react-core'
+import { DefaultPlanSelect } from 'Plans'
+import { Spinner } from 'Common'
+
+import type { Product, ApplicationPlan } from 'Types'
+
+import './DefaultPlanSelectCard.scss'
+
+export type Props = {
+  product: Product,
+  initialDefaultPlan: ApplicationPlan | null,
+  path: string
+}
+
+const NO_DEFAULT_PLAN: ApplicationPlan = { id: -1, name: '(No default plan)' }
+
+const DefaultPlanSelectCard = ({ product, initialDefaultPlan, path }: Props) => {
+  const [defaultPlan, setDefaultPlan] = useState<ApplicationPlan>(initialDefaultPlan ?? NO_DEFAULT_PLAN)
+
+  const [isLoading, setIsLoading] = useState(false)
+
+  const onSelectPlan = (plan: ApplicationPlan) => {
+    const body = plan.id >= 0 ? new URLSearchParams({ id: plan.id.toString() }) : undefined
+    const url = path.replace(':id', String(product.id))
```

Comment:
A bit dangerous to treat it like a string. Maybe we could make path of type URL?

Reply:
In this case path is "guaranteed" to be a string but in any case, URL can't be used for paths but for a full URL, which we don't know.